libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -369,8 +387,14 @@ void AP_BattMonitor::check_failsafes(void)
                     break;
             }
 
-            gcs().send_text(MAV_SEVERITY_WARNING, "Battery %d is %s %.2fV used %.0f mAh", i + 1, type_str,
-                            (double)voltage(i), (double)consumed_mah(i));
+            if (has_cell_voltages(i)) {
+                gcs().send_text(MAV_SEVERITY_WARNING, "Batt %d %s; %.2fV; Used %.0fmAh; Cell %.2fV Diff %.2fV; ", i + 1, type_str,
```

Comment:
This string is much to long, the entire formatted string needs to fit in 50 characters. That should be computed using the longest failsafe type (`critical`) and a battery voltage >10V, and used capacity >1000 mAh (>10k is better).

Reply:
Reverted. Will think of something else.